service/src/main/java/com/epam/esm/servise/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+
+public class UserServiceImpl implements UserService<User> {
+
+    private final UserRepositoryImpl repository;
+    private final CreateUserModelMapper readMapper;
+    private final UserModelReadMapper modelReadMapper;
+    private final CertificateServiceImpl certificateService;
+    private final CreateOrderMapper orderMapper;
+    private final CreateCertificateFromModelCertificateMapper certificateMapper;
+    private final OnlyCertificateReadMapper onlyCertificateReadMapper;
+    private final OrderRepositoryImpl orderRepository;
+    private final ModelCertificateInOnlyCertificateMapper onlyCertificateMapper;
+    private final LanguageMassage languageMassage;
+
+    @Transactional
+    @Override
+    public List<UserModel> getAllEntity(int limit, int offset) {
+        return readMapper.buildUserModelReadMapper(repository.getAllEntity(limit, offset));
+    }
+
+    @Transactional
+    @Override
+    public void saveEntity(User user) {
+        if (repository.getUserByName(user.getNickName()).isEmpty()) {
+            repository.addEntity(user);
+        } else {
+            throw new IncorrectDataException(languageMassage.getMessage("message.such.user"));
+        }
+
+    }
+
+    @Transactional
+    @Override
+    public void updateEntity(long id, User user) {
+        Optional<User> user1 = repository.getEntityById(id);
+        if (user1.isPresent()) {
+            user.setId(id);
+            if(user.getNickName()==null)
+                user.setNickName(user1.get().getNickName());
+            if(user.getEmail()==null)
+                user.setEmail(user1.get().getEmail());
+            repository.updateEntity(user);
+        } else {
+            throw new NoSuchEntityException(languageMassage.getMessage("message.user.with.id"));
+        }
+    }
+
+    @Transactional
+    @Override
+    public Optional<UserModel> getEntity(long id) {
+        Optional<User> user = Optional.ofNullable(repository.getEntityById(id)).orElseThrow();
+        if (user.isEmpty()) {
+            throw new NoSuchEntityException(languageMassage.getMessage("message.user.with.id"));
+        }
+        return user.map(readMapper::mapFrom);
+    }
+
+    @Transactional
+    @Override
+    public void deleteEntity(long id) {
+        Optional<User> user = repository.getEntityById(id);
+        if (user.isEmpty()) {
+            throw new NoSuchEntityException(languageMassage.getMessage("message.user.with.id"));
+        } else {
+            repository.deleteEntity(id);
+        }
+    }
+
+    @Transactional
+    public int countAllUsers() {
+        return repository.countAllUsers();
+    }
+
+    @Transactional
+    public Optional<ReadUserModel> getUserByName(String name) {
+        Optional<User> user = Optional.ofNullable(repository.getUserByName(name)).orElseThrow();
+        return user.map(modelReadMapper::mapFrom);
+    }
+
+
+    @Transactional
+    public CreateOrderModel purchaseCertificate(long userId, long certificateId) {
```

Comment:
all public interfaces should be defined in interface, which represents class

Reply:
made